### PR TITLE
list of accepted meta data for custom Maps

### PR DESCRIPTION
**Pull request by abit2**
https://github.com/sunpy/sunpy/issues/1832 
updated the docstrings of the properties of what meta kwargs they use

### CODE REVIEW

**Comment by abit2** [comment]
Ping @Cadair 
Does it require more documentation?

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -601,6 +655,7 @@ def rotation_matrix(self):
         """
         Matrix describing the rotation required to align solar North with
         the top of the image.
+        Meta keyword : PC1_1, PC1_2, PC2_1, PC2_2, CD1_1, CD1_2, CD2_1, CD2_2
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: some mention of of PC or CD would be nice here.

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -561,19 +604,28 @@ def heliographic_latitude(self):
 
     @property
     def heliographic_longitude(self):
-        """Heliographic longitude"""
+        """
+        Heliographic longitude
+        Meta keyword : hgln_obs
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: for cases like this where a a not-None default is set it would be nice to have that in the docstring as well.

Comment (Cadair) [inline sunpy/map/mapbase.py]: something like `meta keyword: crval1 (default 0 deg), crval2 (default: 0 deg)`

File: sunpy/map/mapbase.py [hunk 3]
```diff
@@ -545,7 +583,12 @@ def carrington_longitude(self):
 
     @property
     def heliographic_latitude(self):
-        """Heliographic latitude"""
+        """
+        Heliographic latitude
+        Meta keyword : hglt_obs
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: this is not very clear, something like `meta keyword: hglt_obs / crlt_obs / solar_b0 [in order of preference]`

**Comment by Cadair** [comment]
Also, all the files in `sunpy/map/sources` should be check to see if they modify any of these properties because then the docstrings should be changed there as well.

**Comment by abit2** [comment]
didn't understand changing sunpy/map/sources part 
can you please explain.

**Comment by abit2** [comment]
@Cadair As far as I can tell, it doesnot change anywhere else in sunpy/map/sources

**Comment by Cadair** [comment]
@abit2 I mean methods like this: https://github.com/sunpy/sunpy/blob/master/sunpy/map/sources/sdo.py#L51 also need documenting in the same fashion.

**Comment by Cadair** [comment]
Sweet Jesus that's a lot of pep8

**Comment by OrkoHunter** [comment]
@abit2 I think the PR needs a rebase to ignore all the trivial E501s. :)

**Comment by Cadair** [comment]
@OrkoHunter I thought it pulled the config from the main repo not the PR? Has that changed since the out-of-date version I was playing with?

**Comment by OrkoHunter** [comment]
@Cadair Earlier it used to pull the config from the main branch. But after testing on different projects, similar to Travis CI, pulling the (updated) config from the PR made more sense.

Now I think, there should be a change. If the config is not there in the PR's branch, it should check the main repo. I'd love to hear your suggestions.

File: sunpy/map/mapbase.py [hunk 4]
```diff
@@ -565,7 +592,15 @@ def rsun_obs(self):
     def coordinate_system(self):
         """
         Coordinate system used for x and y axes (ctype1/2)
-        Meta keyword : ``ctype1``/ ``HPLN-TAN``, ``ctype2``/ ``HPLT-TAN``.
+        .. rubric:: Metadata keywords:
+
+        * First available:
+         1. ``ctype1``
+         2. ``HPLN-TAN``
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: These are defaults not keys

File: sunpy/map/mapbase.py [hunk 5]
```diff
@@ -663,9 +721,16 @@ def rotation_matrix(self):
         """
         Matrix describing the rotation required to align solar North with
         the top of the image.
-        Meta keyword : ``PC1_1``, ``PC1_2``, ``PC2_1``, ``PC2_2``, ``CD1_1``, ``CD1_2``, ``CD2_1``, ``CD2_2``.
-                        returns np matrix with ``PC1_1``, ``PC1_2``, ``PC2_1``, ``PC2_2``
-                        else with ``CD1_1``, ``CD1_2``, ``CD2_1``, ``CD2_2``.
+         .. rubric:: Metadata keywords:
+
+        * ``PC1_1``
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: These are generally written as `PCi_j` and `CDi_j` also these are an either or kinda thing, i.e.
```
* First Available
  1. `PCi_j`
  2. `CDi_j`
```

File: sunpy/map/sources/soho.py [hunk 6]
```diff
@@ -74,8 +74,11 @@ def __init__(self, data, header, **kwargs):
     @property
     def rsun_obs(self):
         """
-        Meta keyword : ``solar_r``, ``cdelt1``.
         Returns the solar radius as measured by EIT in arcseconds.
+        .. rubric:: Metadata keyword:
+
+        * ``solar_r``
```
Comment (Cadair) [inline sunpy/map/sources/soho.py]: This is an actual operation on two keywords so we should probably write it as ``solar_r * cdelt1 `` to make it clear that's what it is.

File: sunpy/map/mapbase.py [hunk 7]
```diff
@@ -211,12 +213,12 @@ def __repr__(self):
 scale:\t\t {scale}
 
 """.format(dtype=self.__class__.__name__,
-           obs=self.observatory, inst=self.instrument, det=self.detector,
-           meas=self.measurement, wave=self.wavelength, date=self.date,
-           dt=self.exposure_time,
-           dim=u.Quantity(self.dimensions),
-           scale=u.Quantity(self.scale),
-           tmf=TIME_FORMAT) + self.data.__repr__())
+                obs=self.observatory, inst=self.instrument, det=self.detector,
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: This change seems to mis-align these kwargs?

File: sunpy/map/mapbase.py [hunk 8]
```diff
@@ -381,7 +383,12 @@ def nickname(self, n):
 
     @property
     def date(self):
-        """Image observation time"""
+        """
+        Image observation time
+        .. rubric:: Metadata keyword:
+
+        * ``date-obs`` Default: ``now``
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: Default ``'now'`` to indicate it is a string.

File: sunpy/map/mapbase.py [hunk 9]
```diff
@@ -392,12 +399,22 @@ def date(self):
 
     @property
     def detector(self):
-        """Detector name"""
+        """
+        Detector name
+        .. rubric:: Metadata keyword:
+
+        * ``detector`` Default: ""
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: Default ``""``

File: sunpy/map/mapbase.py [hunk 10]
```diff
@@ -392,12 +399,22 @@ def date(self):
...
         return self.meta.get('detector', "")
 
     @property
     def dsun(self):
-        """The observer distance from the Sun."""
+        """
+        The observer distance from the Sun.
+        .. rubric:: Metadata keyword:
+
+        * ``dsun_obs`` Default: None
```
Comment (Cadair) [inline sunpy/map/mapbase.py]:
```
Default `None`
```
(single quotes because it's a Python object)

File: sunpy/map/mapbase.py [hunk 11]
```diff
@@ -411,29 +428,59 @@ def dsun(self):
 
     @property
     def exposure_time(self):
-        """Exposure time of the image in seconds."""
+        """
+        Exposure time of the image in seconds.
+        .. rubric:: Metadata keyword:
+
+        * ``exptime`` Default: 0.0
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: backticks

File: sunpy/map/mapbase.py [hunk 12]
```diff
@@ -411,29 +428,59 @@ def dsun(self):
...
         return self.meta.get('exptime', 0.0) * u.s
 
     @property
     def instrument(self):
-        """Instrument name"""
+        """
+        Instrument name
+        .. rubric:: Metadata keyword:
+
+        * ``instrume`` Default: ""
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: backticks

File: sunpy/map/mapbase.py [hunk 13]
```diff
@@ -411,29 +428,59 @@ def dsun(self):
...
         return self.meta.get('instrume', "").replace("_", " ")
 
     @property
     def measurement(self):
-        """Measurement name, defaults to the wavelength of image"""
+        """
+        Measurement name, defaults to the wavelength of image.
+        .. rubric:: Metadata keywords:
+
+        * ``wavelnth`` Default: 0
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: backticks

File: sunpy/map/mapbase.py [hunk 14]
```diff
@@ -411,29 +428,59 @@ def dsun(self):
...
                           self.meta.get('waveunit', ""))
 
     @property
     def wavelength(self):
-        """wavelength of the observation"""
+        """
+        wavelength of the observation
+        .. rubric:: Metadata keyword:
+
+        * ``wavelnth`` Default: 0
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: bacticks

File: sunpy/map/mapbase.py [hunk 15]
```diff
@@ -507,12 +554,26 @@ def shift(self, x, y):
 
     @property
     def rsun_meters(self):
-        """Radius of the sun in meters"""
+        """
+        Radius of the sun in meters
+        .. rubric:: Metadata keyword:
+
+        * ``rsun_ref`` Default: constants.radius
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: this can be a full object link 
```
`sunpy.constants.radius`
```

File: sunpy/map/mapbase.py [hunk 16]
```diff
@@ -507,12 +554,26 @@ def shift(self, x, y):
...
-        """Radius of the Sun."""
+        """
+        Radius of the Sun.
+        ..rubric:: Metadata keyword:
+
+        * First available:
+         1. ``rsun_obs``
+         2. ``solar_r``
+         3. ``radius``
+         4. Default: None
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: single backticks around None

File: sunpy/map/mapbase.py [hunk 17]
```diff
@@ -529,13 +590,29 @@ def rsun_obs(self):
...
                     self.meta.get('ctype2', 'HPLT-TAN'))
 
     @property
     def carrington_longitude(self):
-        """Carrington longitude (crln_obs)"""
+        """
+        Carrington longitude (crln_obs)
+        .. rubric:: Metadata keyword:
+
+        * ``crln_obs`` Default: None
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: backticks

File: sunpy/map/mapbase.py [hunk 18]
```diff
@@ -605,6 +721,16 @@ def rotation_matrix(self):
         """
         Matrix describing the rotation required to align solar North with
         the top of the image.
+         .. rubric:: Metadata keywords:
+
+        * ``PC1_1``
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: These are an either or, and also normally written as `PCi_j` and `CDi_j` to cover all four. i.e it should be a first available with two entries.

**Comment by Cadair** [comment]
@pep8speaks Keep Quiet

**Pull request closed without merging**